linode/framework_provider.go

```diff
@@ -74,6 +74,17 @@ type FrameworkProvider struct {
 	Meta            *helper.FrameworkProviderMeta
 }
 
+func CreateFrameworkProviderWithMeta(version string, meta any) provider.ProviderWithValidateConfig {
+	castMeta := meta.(*helper.ProviderMeta)
+	return &FrameworkProvider{
```

Comment:
Would it be fine to type `meta` as `*helper.ProviderMeta` for the Crossplane provider?

Reply:
And I think it may be more straight forward to pass in a client instead of the whole SDKv2 meta?